src/librepgp/stream-armor.cpp

```diff
@@ -453,29 +453,35 @@ find_armor_header(const char *buf, size_t len, size_t *hdrlen)
     return NULL;
 }
 
+static bool
+str_equals(const char *str, size_t len, const char *another)
+{
+    return (len >= strlen(another)) && !strncmp(str, another, len);
+}
+
 static pgp_armored_msg_t
 armor_str_to_data_type(const char *str, size_t len)
 {
     if (!str) {
         return PGP_ARMORED_UNKNOWN;
     }
-    if (!strncmp(str, "BEGIN PGP MESSAGE", len)) {
+    if (str_equals(str, len, "BEGIN PGP MESSAGE")) {
```

Comment:
What's wrong with `strncmp`? It should react on NULL character on both strings.

Reply:
`strncmp()` will return 0 if str is shorter than "BEGIN PGP MESSAGE", or even empty.

Reply:
@ni4 Not sure what you mean.
These tests pass:
```
    assert_int_not_equal(0, strncmp("aaa", "aaaa", 4));
    assert_int_not_equal(0, strncmp("aaaa", "aaa", 4));
    assert_int_equal(0, strncmp("aaaa", "aaaa", 4));
```